Awesome Book: With ES6 Modules.

maximoortelli opened the pull request:
Awesome Books: With ES6 Modules. In this project, i will restructure my Awesome books app code. The goal is to make it more organized by using modules. I will also practice the ES6 syntax that i have learned.. The goal here is to master all of the tools and best practices learned about in previous steps to use them in all Microverse projects and most likely in my future job as well.

maximoortelli committed:
Add all files

Add all linters
Add index.js
Add index.css
Add directory modules
Add sections.js
Add config.js
Add library Luxon

maximoortelli committed:
Update README.md

Update README.md

maximoortelli committed:
Update README.md

Update README.md

maximoortelli committed:
Update README.md

Update README.md

jerryowusu commented on modules/sections.js:
```diff
@@ -0,0 +1,41 @@
+const section = () => {
+  document.addEventListener('DOMContentLoaded', () => {
+    const $first = document.querySelector('.first');
+    const $showLib = document.querySelector('.showLib');
+    const $titulo = document.querySelector('.titleawe');
+    const $Addnew = document.querySelector('.addnew');
+    const $plus = document.querySelector('.plus');
+    const $cont = document.getElementById('Button');
+    const $Information = document.querySelector('.Inf');
+    const $foot = document.querySelector('.myfoot');
```
- [x] [OPTIONAL] Do not start names with a $  sign since it will put you in conflict with many JavaScript library names. Also, you should try as much as possible to use the camelCase naming convention in order to conform with the JavaScript standard naming convention.  Kindly check this [link](https://www.w3schools.com/js/js_conventions.asp) for more details on that.

jerryowusu commented on modules/time.js:
```diff
@@ -0,0 +1,12 @@
+import { DateTime } from './config.js';
```
According to the project requirements, instead of copying and pasting the luxon files, you are supposed to download the library into your project using `npm` and import it as a module.  eg: `import { DateTime } from "luxon"`. 
Kindly update your project accordingly :+1: 


![Screenshot from 2023-08-21 20-44-40](https://github.com/maximoortelli/Awesome-BookES6/assets/65233860/0b9a580b-1b93-4645-8365-8d01ccc097cb)

maximoortelli replied:
Thank you very much for suggesting I have already made all the requested changes.

jerryowusu reviewed (changes requested):
Hi @maximoortelli 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

![so-close](https://github.com/maximoortelli/Awesome-BookES6/assets/65233860/611b6dc4-5140-4f26-8c1e-7bae472811ab)


## Highlights
 There is an index.html file. :+1: 
 There is an index.js file. :+1: 
 There is an index.css file. :+1: 
 There is a `modules` directory with other .js files. :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

maximoortelli committed:
Update files

Update index.js
Update section.js

maximoortelli committed:
Merge branch 'feature-modules' of https://github.com/maximoortelli/Awesome-BookES6 into feature-modules

maximoortelli committed:
Update all files

I download the library in my project using npm and import it as a module.

I change all the name with the $ sign.

1GORDON reviewed (approved):
# Status: APPROVED 🟢 

Hi @maximoortelli ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Added the library into your project using npm ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
